File: iosApp/iosApp/Libs/Infrastructure/Network/Auth/Interface/AuthClient.swift
import Foundation

protocol AuthClient {
  var identity: DelegatedIdentity? { get }
  var canisterPrincipal: Principal? { get }
  var canisterPrincipalString: String? { get }
  var userPrincipal: Principal? { get }
  var userPrincipalString: String? { get }
  func initialize() async throws
  func refreshAuthIfNeeded(using cookie: HTTPCookie) async throws
  func generateNewDelegatedIdentity() throws -> DelegatedIdentity
  func generateNewDelegatedIdentityWireOneHour() throws -> DelegatedIdentityWire
}

File: shared/rust/src/commonMain/kotlin/com/yral/shared/rust/auth/AuthClient.kt
package com.yral.shared.rust.auth

import com.yral.shared.uniffi.generated.Principal
import io.ktor.client.plugins.cookies.HttpCookies

interface AuthClient {
    var identity: ByteArray?
    var canisterPrincipal: Principal?
    var userPrincipal: Principal?

    suspend fun initialize()
    suspend fun refreshAuthIfNeeded(cookie: HttpCookies)
    suspend fun generateNewDelegatedIdentity(): ByteArray
    suspend fun generateNewDelegatedIdentityWireOneHour(): ByteArray
}
